src/parser/DASHTree.cpp

```diff
@@ -236,17 +236,8 @@ bool ParseContentProtection(const char** attr, DASHTree* dash)
 
 static void ReplacePlaceHolders(std::string& rep, const std::string& id, uint32_t bandwidth)
 {
-  std::string::size_type repPos = rep.find("$RepresentationID$");
-  if (repPos != std::string::npos)
-    rep.replace(repPos, 18, id);
-
-  repPos = rep.find("$Bandwidth$");
-  if (repPos != std::string::npos)
-  {
-    char bw[32];
-    sprintf(bw, "%u", bandwidth);
-    rep.replace(repPos, 11, bw);
-  }
+  replaceAll(rep, "$RepresentationID$", id, true);
```

Comment:
if you set nextEmpty to true only the first case will be replaced all others will be deleted